var/spack/repos/builtin/packages/py-onnx/package.py

```diff
@@ -46,12 +55,18 @@ class PyOnnx(PythonPackage):
     depends_on("py-protobuf@3.12.2:3.20.1", type=("build", "run"), when="@1.12.0")
     depends_on("protobuf@3.20.2:3", when="@1.13.0:")
     depends_on("py-protobuf@3.20.2:3", type=("build", "run"), when="@1.13.0:")
+
     depends_on("py-numpy", type=("build", "run"))
     depends_on("py-numpy@1.16.6:", type=("build", "run"), when="@1.8.1:")
-    depends_on("py-six", type=("build", "run"), when="@:1.8.1")
-    depends_on("py-typing-extensions@3.6.2.1:", type=("build", "run"))
+
+    # CMakeLists.txt
     depends_on("cmake@3.1:", type="build")
-    depends_on("py-pytest-runner", type="build")
+    depends_on("py-pybind11@2.2:", type=("build", "run"))
```

Comment:
It's needed at run-time too? Or should it be type link?

Reply:
Both `onnx` and `py-onnx` install a C++ header `py_utils.h` which includes `pybind11.h`. I thought that makes it run-type, not just link-type, right?

Reply:
In any case, `onnx` should get the same depends_on after we decide on the type.

Reply:
(and `pybind11` is header only so there is no library.)

Reply:
Link type means use Spack's compiler wrappers for the libs/headers. Run type means add to PATH and PYTHONPATH at run time. Run doesn't mean anything for header-only libraries.

Reply:
```suggestion
    depends_on("py-pybind11@2.2:", type=("build", "link"))
```